### ISSUE TITLE
Error when date has different format

### ISSUE THREAD

**Issue by rEnr3n** [soywod/himalaya#369]
```
$ himalaya --version
himalaya 0.5.10
```

I'm using the maildir backend and I got this error when I start the app.

```
$ himalaya
Error: cannot parse maildir envelopes from "/home/ren/.mail/purelymail"

Caused by:
    0: cannot parse maildir mail entry
    1: cannot parse maildir message date "05-FEB-2021 22:05:02"
    2: input contains invalid characters
```

I checked the email and it has the following three lines:
```
Received: from smtp-nlb02 ([10.141.249.58]) by M61SV059 with Microsoft SMTPSVC(10.0.17763.1);
	 Fri, 5 Feb 2021 22:05:02 +0800
Date: 05-FEB-2021 22:05:02
```

I don't see this in other emails. The other emails have the following format: `Date: Fri, 5 Feb 2021 22:05:02 +0800`

**Comment by soywod** [comment]
Clearly the date does not follow the [RFC](https://datatracker.ietf.org/doc/html/rfc2822#section-3.3). I cannot even see from which RFC this date format derives from :disappointed: not sure how to help. Any suggestion?

**Comment by rEnr3n** [comment]
I checked all the emails that have problematic values on `Date:`. They all have the correct format at the end of `Received:` just like OP. Others have these:

```
Received: from smtp-nlb02 ([10.141.249.58]) by M61SV059 with Microsoft SMTPSVC(10.0.17763.1)
    by mail.example.com
    ; Fri, 5 Feb 2021 22:05:02 +0800
```

```
Received: from smtp-nlb02 ([10.141.249.58]) by M61SV059 with Microsoft SMTPSVC(10.0.17763.1)
    by mail.example.com
    for <contact@example.com>; Fri,  5 Feb 2021 22:05:02 +0800
```

I don't completely understand the RFC but there's a mention [here](https://datatracker.ietf.org/doc/html/rfc2822#section-2.2.3) about folding. Is that what you're looking for?

**Comment by soywod** [comment]
> I don't completely understand the RFC but there's a mention [here](https://datatracker.ietf.org/doc/html/rfc2822#section-2.2.3) about folding. Is that what you're looking for?

It could be the issue indeed. I will investigate. Could you share a complete raw email so I can reproduce and debug locally?

**Comment by rEnr3n** [comment]
I just sent you an email with the attached file.

**Comment by soywod** [comment]
Fixed by [6bd26f5](https://git.sr.ht/~soywod/himalaya-lib/commit/6bd26f512585047c6914def048d7e5ff6a824a33), should be added to the next release #433.
